server/lib/availabilitySync.ts

```diff
@@ -234,62 +183,115 @@ class AvailabilitySync {
     } while (mediaPage.length > 0);
   }
 
+  private async findMediaStatus(
+    requests: MediaRequest[],
+    is4k?: boolean
+  ): Promise<MediaStatus> {
+    let filteredRequests: MediaRequest[] = requests;
+
+    if (is4k !== undefined) {
+      filteredRequests = requests.filter((request) => request.is4k === is4k);
+    }
+
+    let mediaStatus: MediaStatus;
+
+    if (
+      filteredRequests.some(
+        (request) => request.status === MediaRequestStatus.APPROVED
+      )
+    ) {
+      mediaStatus = MediaStatus.PROCESSING;
+    } else if (
+      filteredRequests.some(
+        (request) => request.status === MediaRequestStatus.PENDING
+      )
+    ) {
+      mediaStatus = MediaStatus.PENDING;
+    } else {
+      mediaStatus = MediaStatus.UNKNOWN;
+    }
+
+    return mediaStatus;
+  }
+
   private async mediaUpdater(media: Media, is4k: boolean): Promise<void> {
     const mediaRepository = getRepository(Media);
     const requestRepository = getRepository(MediaRequest);
 
-    const isTVType = media.mediaType === 'tv';
-
     try {
-      const request = await requestRepository.findOne({
-        relations: {
-          media: true,
-        },
-        where: { media: { id: media.id }, is4k: is4k ? true : false },
-      });
-
       logger.info(
-        `Media ID ${media.id} does not exist in your ${
+        `Media with TMDB ID ${media.tmdbId} does not exist in your ${
```

Comment:
All logs that showed media id will now show the tmdbId instead. This will allow better debugging.